hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java

```diff
@@ -316,6 +317,18 @@ public DirectoryScanner(FsDatasetSpi<?> dataset, Configuration conf) {
 
     masterThread =
         new ScheduledThreadPoolExecutor(1, new Daemon.DaemonFactory());
+
+    reconcileBlocksBatchSize =
+        conf.getInt(DFSConfigKeys.
+                DFS_DATANODE_RECONCILE_BLOCKS_BATCH_SIZE,
+            DFSConfigKeys.
+                DFS_DATANODE_RECONCILE_BLOCKS_BATCH_SIZE_DEFAULT);
+
+    reconcileBlocksBatchInterval =
+        conf.getInt(DFSConfigKeys.
+                DFS_DATANODE_RECONCILE_BLOCKS_BATCH_INTERVAL,
+            DFSConfigKeys.
+                DFS_DATANODE_RECONCILE_BLOCKS_BATCH_INTERVAL_DEFAULT);
```

Comment:
Add a validation for these configs, if ``reconcileBlocksBatchSize`` and ``reconcileBlocksBatchInterval`` is less than one use default. and add a warn log message if these values are incorrect something like:
Invalid value configured for < config name>, should be greater than 0, Using default.

In the end add an Info log for the values being used.

Reply:
Fixed in latest PR.